plugins/modules/fusion_volume.py

```diff
@@ -326,9 +326,10 @@ def update_size(module, fusion, current, patches):
 
 def update_protection_policy(module, fusion, current, patches):
     wanted = module.params
+    current_policy = current.protection_policy.name if current.protection_policy else ""
     if (
-        wanted["protection_policy"]
-        and wanted["protection_policy"] != current.protection_policy.name
+        wanted["protection_policy"] is not None
+        and wanted["protection_policy"] != current_policy
     ):
         patch = purefusion.VolumePatch(
             protection_policy=purefusion.NullableString(wanted["protection_policy"])
```

Comment:
If we want to unset the protection policy, shouldn't we pass None value rather than the empty string? (honestly, I'm not sure if Fusion would accept that) Putting empty string into NullableString when wanting to unset something looks weird to me.

At the same time, I don't think we can pass None value in Ansible playbooks... maybe if you omit the field at all? But then we would use Ansible like a terraform... what do you think? @jankodera-ps

Reply:
Our Ansible modules are generally designed to not touch fields which are not present. Thus in playbook passing empty string for name is the only way to erase it. In most places backend treats empty string and no value the same, because Go mostly also behaves like that, so at the moment this works and keeps the code simple. I would keep it like this for now and if we change a generator, which would mean type hinting, we could probably redesign it to work a bit better. Passing None would add another if.